src/libutils/oglapphelpers/oglapp.cpp

```diff
@@ -264,5 +230,161 @@ void OglApp::printGLInfo() const noexcept
               << "GL Version:   " << glGetString(GL_VERSION) << std::endl
               << "GLSL Version: " << glGetString(GL_SHADING_LANGUAGE_VERSION) << std::endl;
 }
+
+void OglApp::setupCommon()
+{
+
+#ifndef __APPLE__
+    glewInit();
+    // TO DO: Find out why glewInit() != GLEW_OK
+
+    if (!glewIsSupported("GL_VERSION_2_0"))
+    {
+        throw Exception("OpenGL 2.0 not supported.");
+    }
+#endif
+
+    // Initialize the OpenGL engine.
+
+    glPixelStorei(GL_UNPACK_ALIGNMENT, 4);           // 4-byte pixel alignment
+
+#ifndef __APPLE__
+    glClampColor(GL_CLAMP_READ_COLOR, GL_FALSE);     //
+    glClampColor(GL_CLAMP_VERTEX_COLOR, GL_FALSE);   // avoid any kind of clamping
+    glClampColor(GL_CLAMP_FRAGMENT_COLOR, GL_FALSE); //
+#endif
+
+    glEnable(GL_TEXTURE_2D);
+}
+
+OglAppRcPtr OglApp::CreateOglApp(const char * winTitle, int winWidth, int winHeight)
+{
+#ifdef OCIO_HEADLESS_ENABLED
+        return std::make_shared<HeadlessApp>(winTitle, winWidth, winHeight);
+#else
+        return std::make_shared<ScreenApp>(winTitle, winWidth, winHeight);
+#endif
+}
+
+ScreenApp::ScreenApp(const char * winTitle, int winWidth, int winHeight):
+    OglApp(winWidth, winHeight)
+{
+    int argc = 2;
+    const char * argv[] = { winTitle, "-glDebug" };
+
+    glutInit(&argc, const_cast<char**>(&argv[0]));
+
+    glutInitDisplayMode(GLUT_RGBA | GLUT_DOUBLE | GLUT_DEPTH);
+    glutInitWindowSize(m_viewportWidth, m_viewportHeight);
+    glutInitWindowPosition(0, 0);
+
+    m_mainWin = glutCreateWindow(argv[0]);
+
+    setupCommon();
+}
+
+ScreenApp::~ScreenApp()
+{
+    glutDestroyWindow(m_mainWin);
+}
+
+void ScreenApp::redisplay()
+{
+    OglApp::redisplay();
+    glutSwapBuffers();
+}
+
+void ScreenApp::printGLInfo() const noexcept
+{
+    OglApp::printGLInfo();
+}
+
+#ifdef OCIO_HEADLESS_ENABLED
+
+HeadlessApp::HeadlessApp(const char * winTitle, int bufWidth, int bufHeight)
+    : OglApp(bufWidth, bufHeight)
+    , m_pixBufferWidth(bufWidth)
+    , m_pixBufferHeight(bufHeight)
+{
+    m_configAttribs =
+    {
+              EGL_SURFACE_TYPE, EGL_PBUFFER_BIT,
+              EGL_BLUE_SIZE, 8,
+              EGL_GREEN_SIZE, 8,
+              EGL_RED_SIZE, 8,
+              EGL_DEPTH_SIZE, 8,
+              EGL_RENDERABLE_TYPE, EGL_OPENGL_BIT,
+              EGL_NONE
```

Comment:
Only one extra tabulation (i.e. four spaces as tabulation is banned) is needed like  `m_pixBufferAttribs`.